test/globalHooks.js

```diff
@@ -0,0 +1,30 @@
+const nock = require('nock');
+const chai = require('chai');
+const getSampleResponse = require('./resources/getSampleResponse');
+
+const expect = chai.expect;
+
+before('global setup', () => {
+  const ls27ue = 'LS27UE';
+  const ls27ueResponse = getSampleResponse('postcodesio-responses/ls27ue.json');
+  const serviceApiResponse = getSampleResponse('service-api-responses/-1,54.json');
+  const ls27ueResult = JSON.parse(ls27ueResponse).result;
+  const latitude = ls27ueResult.latitude;
+  const longitude = ls27ueResult.longitude;
+
+  nock(process.env.API_BASE_URL)
+    .get(`/nearby?latitude=${latitude}&longitude=${longitude}`)
+    .times(4)
+    .reply(200, serviceApiResponse);
+
+  nock('https://api.postcodes.io')
+    .get(`/postcodes/${ls27ue}`)
+    .times(4)
+    .reply(200, ls27ueResponse);
+});
+
+after('Check all nocks have been called', () => {
+  expect(nock.pendingMocks().length).to.equal(0);
+  expect(nock.isDone()).to.equal(true);
```

Comment:
That post is about the same thing i.e nock is global. The solution of executing `nock.cleanAll()` is done here too and it might be possible to have the nock setup/teardown done back in the local tests. However, I prefer having it done globally to highlight the fact nock is global. I've also seen nocks still being active once the `cleanAll()` has been run so I have my doubts whether it would definitely work.